test/units/module_utils/common/text/formatters/test_human_to_bytes.py

```diff
@@ -10,61 +10,60 @@
+    'G': 1073741824,
+    'T': 1099511627776,
+    'P': 1125899906842624,
+    'E': 1152921504606846976,
+    'Z': 1180591620717411303424,
+    'Y': 1208925819614629174706176,
+}
+
+
 @pytest.mark.parametrize(
     'input_data,expected',
     [
-        (0, 0),
-        (1024, 1024),
-        (u'1024B', 1024),
-        (u'0B', 0),
-        (u'1K', 1024),
-        (u'1KB', 1024),
-        (u'1MB', 1048576),
-        (u'1M', 1048576),
-        (u'1G', 1073741824),
-        (u'1GB', 1073741824),
-        (u'1T', 1099511627776),
-        (u'1TB', 1099511627776),
-        (u'1P', 1125899906842624),
-        (u'1PB', 1125899906842624),
-        (u'1E', 1152921504606846976),
-        (u'1EB', 1152921504606846976),
-        (u'1Z', 1180591620717411303424),
-        (u'1ZB', 1180591620717411303424),
-        (u'1Y', 1208925819614629174706176),
-        (u'1YB', 1208925819614629174706176),
+        ((0, u'0B'), 0),
+        ((1024, u'1024B'), NUM_IN_METRIC['K']),
+        ((u'1K', u'1KB'), NUM_IN_METRIC['K']),
+        ((u'1M', u'1MB'), NUM_IN_METRIC['M']),
+        ((u'1G', u'1GB'), NUM_IN_METRIC['G']),
+        ((u'1T', u'1TB'), NUM_IN_METRIC['T']),
+        ((u'1P', u'1PB'), NUM_IN_METRIC['P']),
+        ((u'1E', u'1EB'), NUM_IN_METRIC['E']),
+        ((u'1Z', u'1ZB'), NUM_IN_METRIC['Z']),
+        ((u'1Y', u'1YB'), NUM_IN_METRIC['Y']),
     ]
 )
 def test_human_to_bytes_number(input_data, expected):
     """Test of human_to_bytes function, only number arg is passed."""
-    assert human_to_bytes(input_data) == expected
+    for elem in input_data:
+        assert human_to_bytes(elem) == expected
 
 
 @pytest.mark.parametrize(
-    'input_data,unit,expected',
+    'input_data,units',
     [
-        (u'1024', u'B', 1024),
-        (1, u'K', 1024),
-        (1, u'KB', 1024),
-        (u'1', u'M', 1048576),
-        (u'1', u'MB', 1048576),
-        (1, u'G', 1073741824),
-        (1, u'GB', 1073741824),
-        (1, u'T', 1099511627776),
-        (1, u'TB', 1099511627776),
-        (u'1', u'P', 1125899906842624),
-        (u'1', u'PB', 1125899906842624),
-        (u'1', u'E', 1152921504606846976),
-        (u'1', u'EB', 1152921504606846976),
-        (u'1', u'Z', 1180591620717411303424),
-        (u'1', u'ZB', 1180591620717411303424),
-        (u'1', u'Y', 1208925819614629174706176),
-        (u'1', u'YB', 1208925819614629174706176),
+        (u'1024', (u'B', u'B')),
+        (1, (u'K', u'KB')),
+        (u'1', (u'M', u'MB')),
+        (1, (u'G', u'GB')),
+        (1, (u'T', u'TB')),
+        (u'1', (u'P', u'PB')),
+        (u'1', (u'E', u'EB')),
+        (u'1', (u'Z', u'ZB')),
+        (u'1', (u'Y', u'YB')),
     ]
 )
-def test_human_to_bytes_number_unit(input_data, unit, expected):
+def test_human_to_bytes_number_unit(input_data, units):
     """Test of human_to_bytes function, number and default_unit args are passed."""
-    assert human_to_bytes(input_data, default_unit=unit) == expected
+    for unit in units:
+        if unit[0] == u'B':
```

Comment:
Please, don't add any logic into tests. It makes them fragile and unreadable. Separate them.

Reply:
removed